enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js

```diff
@@ -275,6 +275,89 @@ describeWithToken("formatting > sandboxes", () => {
       });
     });
 
+    it.skip("should allow joins to the sandboxed table (metabase-enterprise#154)", () => {
+      const COLLECTION_GROUP_ID = 4;
+      withSampleDataset(
+        ({ PEOPLE, PEOPLE_ID, ORDERS_ID, PRODUCTS_ID, REVIEWS_ID }) => {
+          cy.log(
+            "**-- 1. Sandbox `People` table on `user_id` attribute for `data` group --**",
+          );
+
+          cy.request("POST", "/api/mt/gtap", {
+            attribute_remappings: {
+              user_id: ["dimension", ["field-id", PEOPLE.ID]],
+            },
+            card_id: null,
+            group_id: COLLECTION_GROUP_ID,
+            table_id: PEOPLE_ID,
+          });
+
+          cy.log("**-- 2. Fetch permissions graph --**");
+
+          cy.request("GET", "/api/permissions/graph", {}).then(
+            ({ body: { groups, revision } }) => {
+              // Update permissions for `collections` group [id: 4]
+              // This mutates the original `groups` object => we'll pass it next to the `PUT` request
+              groups[COLLECTION_GROUP_ID] = {
+                1: {
+                  schemas: {
+                    PUBLIC: {
+                      [ORDERS_ID]: "all",
+                      [PEOPLE_ID]: { query: "segmented", read: "all" },
+                      [PRODUCTS_ID]: "all",
+                      [REVIEWS_ID]: "all",
+                    },
+                  },
+                },
+              };
+
+              cy.log("**-- 3. Update/save permissions --**");
+
+              cy.request("PUT", "/api/permissions/graph", {
+                groups,
+                revision,
+              });
+            },
+          );
+        },
+      );
+
+      signOut();
+
+      cy.log("**-- Logging in as sandboxed user --**");
+      cy.request("POST", "/api/session", {
+        username: sandboxed_user.email,
+        password: sandboxed_user.password,
+      });
+
+      // Go straight to orders table in custom questions
+      cy.visit("/question/new?database=1&table=2&mode=notebook");
+      cy.findByText("Summarize").click();
+      cy.findByText("Count of rows").click();
+      cy.findByText("Pick a column to group by").click();
+
+      cy.log(
+        "**-- Original issue reported failure to find 'User' group / foreign key--**",
+      );
+      cy.get(".ReactVirtualized__Grid").scrollTo("bottom"); // Cypress scroll sometimes doesn't work; We should disable virtualization for Popover
+      popover().within(() => {
+        cy.get(".List-section-header")
+          .contains("User")
+          .click();
+      });
+
+      cy.get(".ReactVirtualized__Grid").scrollTo(0, 0); // This is needed again because of the virtualization
+      popover().within(() => {
+        cy.get(".List-item")
+          .contains("ID")
+          .click();
+      });
+
+      cy.findByText("Visualize").click();
+      cy.findByText("Count by User → ID");
+      cy.findAllByText("10");
```

Comment:
Is this sandboxed user with attribute `user_id=3`? Otherwise I'm unsure what the `10` assertion is.

Reply:
It is sandboxed user with the `user_id=1`. Accompanied `count` should be `11`. Great catch and an overlook on my side. I foolishly used the UI from unsandboxed (admin) results.

Thank you.